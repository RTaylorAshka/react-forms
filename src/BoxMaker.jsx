import { useState } from "react";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";

function BoxMaker() {

    const [boxes, setBoxes] = useState([])

    function addBox({ width, height, color }) {
        setBoxes([...boxes, { width, height, color }])
    }

    function deleteBox(idx) {
        setBoxes(boxes.filter((t, i) => i != idx));
    }

    return (
        <div>
            <h1>Box Maker</h1>
            <NewBoxForm addBox={addBox} />
            {boxes.map((b, idx) => <Box width={b.width} height={b.height} color={b.color} deleteBox={() => deleteBox(idx)} key={idx} />)}


        </div>
    )

}

export default BoxMaker;