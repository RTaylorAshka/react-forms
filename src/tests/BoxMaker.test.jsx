import { render, fireEvent, queryByLabelText, queryAllByAttribute } from "@testing-library/react";
import BoxMaker from "../BoxMaker";

it("should render BoxMaker without crashing", () => {
    render(<BoxMaker />);
})

it("matches snapshot", () => {
    const { asFragment } = render(<BoxMaker />)
    expect(asFragment()).toMatchSnapshot();
})

it("should add a new box", () => {
    const { container } = render(<BoxMaker />)
    let boxes = container.querySelectorAll('div.box')
    expect(boxes.length).toEqual(0);


})