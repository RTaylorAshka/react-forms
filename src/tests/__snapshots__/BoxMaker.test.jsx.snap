// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Box Maker
    </h1>
    <form>
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        max="7"
        min="3"
        name="width"
        type="range"
        value="3"
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        max="7"
        min="3"
        name="height"
        type="range"
        value="3"
      />
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        type="color"
        value="#e66465"
      />
      <button>
        create
      </button>
    </form>
  </div>
</DocumentFragment>
`;
